my-app/src/index1.js

```diff
@@ -0,0 +1,291 @@
+      squares: squares,
+    };
+
+    squaresHistory.push(history);
+
+    const winner = calculateWinner(squares);
+
+    this.setState({
+      squares: squares,
+      squaresHistory: squaresHistory,
+      xIsNext: xIsNext,
+      winner: winner,
+      showHistoryStep: squaresHistory.length - 1,
+    });
+  }
+
+  handleHistoryClick(index) {
+    this.setState({ showHistoryStep: index });
+  }
+
+  render() {
+    return (
+      <div className="game">
+        <div className="game-board">
+          <Board
+            winner={this.state.winner}
+            squares={this.state.squares}
+            squaresHistory={this.state.squaresHistory}
+            showHistoryStep={this.state.showHistoryStep}
+            onClick={(index) => this.handleClick(index)}
+          ></Board>
+        </div>
+        <div className="game-info">
+          <PlayerStatus
+            xIsNext={this.state.xIsNext}
+            winner={this.state.winner}
+          ></PlayerStatus>
+          <div className="choosed-options">
+            <ChossedOptions
+              squaresHistory={this.state.squaresHistory}
+              onClick={(index) => this.handleHistoryClick(index)}
+            ></ChossedOptions>
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+function PlayerStatus(props) {
+  let playerStatus = props.xIsNext ? "Next player: X" : "Next player: O";
+
+  if (props.winner != null) {
+    playerStatus = "The winner is: " + props.winner + " !!!";
+  }
+
+  return <div className="status">{playerStatus}</div>;
+}
+
+function ChossedOptions(props) {
+  const historyOptions = props.squaresHistory.slice();
+
+  let index = -1;
+  var historyElements = historyOptions.map((element) => {
+    if (element != null) {
+      index++;
+      return (
+        <HistoryOption
+          index={index}
+          onClick={(index) => props.onClick(index)}
+        ></HistoryOption>
+      );
+    }
+    return "";
+  });
+
+  return historyElements;
+}
+
+function HistoryOption(props) {
+  let buttonText = "";
+
+  if (props.index === 0) {
+    buttonText += "Go to game start";
+  } else {
+    buttonText += "Go to move #" + props.index + ".";
+  }
+
+  return (
+    <div>
+      <p>
+        {props.index + 1 + ". "}{" "}
+        <button
+          className="history-button"
+          onClick={(index) => props.onClick(props.index)}
+        >
+          {buttonText}
+        </button>
+      </p>
+    </div>
+  );
+}
+
+function RenderSquareHistory(props) {
+  const value = props.squares.squares[props.index];
+
+  if (value !== null) {
+    return <Square value={value} onClick={props.onClick} />;
+  } else {
+    return <Square value="" onClick={props.onClick} />;
+  }
+}
+
+function RenderSquare(props) {
+  return (
+    <Square
+      value={props.squares[props.index]}
+      onClick={(index) => props.onClick(props.index)}
+    />
+  );
+}
+
+function Square(props) {
+  return (
+    <button className="square" onClick={props.onClick}>
```

Comment:
Wait so at the end you are not sending the index as a param of the onclick. So where is the index param in <Game> compoent 'handleClick' fn coming from?